aten/src/ATen/native/quantized/cpu/fused_obs_fake_quant.cpp

```diff
@@ -129,6 +129,62 @@ std::tuple<at::Tensor, at::Tensor> choose_qparams_fake_quant(
 namespace at {
 namespace native {
 
+std::tuple<
+    at::Tensor,
+    at::Tensor,
+    at::Tensor,
+    at::Tensor,
+    at::Tensor,
+    at::Tensor
+> fused_moving_avg_obs_fake_quant_functional(
+    const at::Tensor& self,
+    const at::Tensor& observer_on,
+    const at::Tensor& fake_quant_on,
+    const at::Tensor& running_min,
+    const at::Tensor& running_max,
+    const at::Tensor& scale,
+    const at::Tensor& zero_point,
+    const double averaging_const,
+    const int64_t quant_min,
+    const int64_t quant_max,
+    const int64_t ch_axis,
+    bool per_row_fake_quant,
+    bool symmetric_quant) {
+  // make local copies of all of the mutable inputs
+  auto running_min_copy = running_min.clone();
+  auto running_max_copy = running_max.clone();
+  auto scale_copy = scale.clone();
+  auto zero_point_copy = zero_point.clone();
+  // call into the mutable version of the op
+  auto out = at::_fused_moving_avg_obs_fq_helper(
+    self,
+    observer_on,
+    fake_quant_on,
+    // Careful - the functional and non-functional version of this operator have the same C++ API name.
+    // The only difference in their C++ schemas is that the non-functional version takes in non-const tensors.
+    // (If we call into the funtional version, we'll infinite loop).
```

Comment:
This feels kinda unfortunate. I figure it's alright though because:
- This comes up only in very uncommon cases specific to functionalization
- We're already tied in to using `const Tensor` vs. `Tensor` as a way to distinguish function schemas in other parts of the codebase (I remember this coming up in the faithful vs. nonfaithful C++ API overloads, where the out= argument can come either first or last).

Reply:
IMO, this seems bad, especially because we want to STOP using mutable arguments to distinguish between mutable and non-mutable inputs. I think we should come up with some naming convention (similar to how we do `_out` for out overloads) to make sure these are distinguished in C++.

Reply:
That seems reasonable (although I feel a little bad adding some special handling just for functionalization in the code that spits out C++ API names).

What do you think of `foo.functional` -> we check if the overload name is `functional` and make the name `foo_functional`? (Basically specializing on both `out` and `functional` now, instead of just `out`)

Reply:
> What do you think of foo.functional -> we check if the overload name is functional and make the name foo_functional? (Basically specializing on both out and functional now, instead of just out)

Yes, this seems reasonable.